woocommerce/woocommerce-gutenberg-products-block
# Link cart/checkout merchant (editor) feedback prompt to special GitHub issue type

Fixes #2917 

This PR adds a new issue template for cart & checkout feedback, and links to it from the editor `Feedback` inspector panel.

<img width="308" alt="Screen Shot 2020-08-06 at 3 11 26 PM" src="https://user-images.githubusercontent.com/4167300/89486492-515a7c80-d7f7-11ea-9711-21aed4bc23bb.png">

Note - the issue template is brief to encourage more feedback - don't want the template to feel like a heavy form "barrier". 

#### tidied issue templates
I've added `labels` to metadata, and also done this for our existing templates. This should automatically label new issues - I'll test to check this is working correctly after merge. Keen for feedback on which labels to use for each template (or if this is overkill).

Also I changed all the headings to use markdown heading format (e.g. `## Heading`) rather than bold. This is more semantically correct/clear, more compact, and easier to edit (bold has start/end markers; heading is simpler, just hashes on start of line). This also allows us to use heading hierarchy if that's useful (added to `Environment` section). A bit sneaky adding in this PR but I figure let's make these consistent while here - happy to revert this if team prefers other format.

#### `withFeedbackPrompt` HOC and existing component

As part of this I noticed that the `withFeedbackPrompt` HOC is no longer used, so I've deleted it. 

Also, the existing FeedbackPrompt links to the ideas board. I've retained this default behaviour, though it's not used - could simply remove it, focus specifically on cart/checkout (since that's the only use case for this component). 


### How to test the changes in this Pull Request:

- Read through the new [issue template](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/894062b53ca1d3f50fb95a75619f2ca752f524f5/.github/ISSUE_TEMPLATE/--cart-checkout-feedback.md). Does it make you want to provide clear useful feedback?
- Edit a page with cart or checkout block, confirm the inspector prompt is enticing and friendly (and links to the right place).
